--- a/src/osx/cocoa/utils_base.mm	2019-12-15 13:02:59.224595546 +0100
+++ b/src/osx/cocoa/utils_base.mm	2019-12-15 13:04:35.110432000 +0100
@@ -215,10 +215,18 @@
     {
         wxLogDebug(wxT("wxCocoaLaunch Can't open path: %s"), path.c_str());
         return false ;
     }
     
+    // Don't try running non-bundled applications here, we don't support output
+    // redirection, which is important for them, unlike for the bundled apps,
+    // so let the generic Unix code handle them.
+    if ( [NSBundle bundleWithURL:url] == nil )
+    {
+        return false;
+    }
+
     // Loop through command line arguments to the bundle,
     // turn them into CFURLs and then put them in cfaFiles
     // For use to launch services call
     NSMutableArray *params = [[NSMutableArray alloc] init];
     for( ; *argv != NULL; ++argv )
